### ISSUE TITLE
xesmf doesn't work in pangeo-notebook

### ISSUE THREAD

**Issue by rabernat** [pangeo-data/pangeo-docker-images#101]
As reported on discourse: https://discourse.pangeo.io/t/regridding-error-while-using-xesmf-mpi-module-missing/749

**Comment by dcherian** [comment]
You have to do something special to get the right environment with MPI. @ocefpaf helped me with this on the pangeo gitter a while ago.

**Comment by ocefpaf** [comment]
You need to request the `mpi` variant you want with `"mpi=*=openmpi"`. Conda is not smart enough, or maybe it is to allow multiple mpis?

```
conda create --name TEST xesmf "mpi=*=openmpi"
```

If you want mpich you can do,

```
conda create --name TEST xesmf "mpi=*=mpich"
```

Note that the solver may choke on packages that are not built with the desired mpi implementation.

**Comment by scottyhq** [comment]
Thanks @ocefpaf these two variants seem to work:

1) conda create --name TEST "mpi=*=openmpi" "esmpy=*=mpi_openmpi*" xesmf
2) conda create --name TEST "mpi=*=mpich" esmpy xesmf

@rabernat or @dcherian which do you recommend? I've not used xesmf before...

**Comment by dcherian** [comment]
i got `mpich` working on my laptop but that's the extent of my knowledge :)

**Comment by ocefpaf** [comment]
> 1. conda create --name TEST "mpi=*=openmpi" "esmpy=*=mpi_openmpi*" xesmf

There is probably a mistake somewhere in the package that made this extra string required. I'll look into it.

**Comment by scottyhq** [comment]
If anyone wants to open a PR with the your favorite xesmf variant we can definitely get it merged.

**Comment by shunzlu** [comment]
Hi everyone, from this issue reply (https://github.com/JiaweiZhuang/xESMF/issues/47#issuecomment-665516640), it seems that we can get MPI to install with esmpy last version, therefore, adding esmpy explicitly into the environmet.yml file list may solve this problem. I've tried it and it works for my pangeo binder, but I am not sure if this also works for pangeo cloud.

**Comment by amv10070** [comment]
Hi all, any chance one of you who understands the xesmf issue better than I could help me with @scottyhq's suggestion to open a PR for fixing this?  I would really like to be able to use xesmf again on the us-central1-b.gcp.pangeo.io cluster

**Comment by TomAugspurger** [comment]
@amv10070 if I understand https://github.com/pangeo-data/pangeo-docker-images/issues/101#issuecomment-670292539 correctly, it sounds like adding `esmpy` to https://github.com/pangeo-data/pangeo-docker-images/blob/master/pangeo-notebook/environment.yml will be all we need. Then bots will rebuild everything else.

**Comment by scottyhq** [comment]
@amv10070 and @shunzlu - please do open a PR with the instructions above ! Then we can get new images on the hubs ASAP.

**Comment by scottyhq** [comment]
closed by #122 so xesmf should work with `pangeo/pangeo-notebook:2020.09.01` and later tags
